subprojects/core/src/main/java/org/gradle/process/internal/DefaultExecHandle.java

```diff
@@ -231,7 +232,7 @@ public ExecHandle start() {
             setState(ExecHandleState.STARTING);
 
             execHandleRunner = new ExecHandleRunner(this, streamsHandler, processLauncher, executorFactory);
-            executor.execute(execHandleRunner);
+            executor.execute(new BuildOperationIdentifierPreservingRunnable(execHandleRunner));
```

Comment:
OK. I think it's probably going to be pretty common that we have a single operation spread out over other threads/workers and need to correlate them all. I'm OK with things as-is.